packages/react-combobox/src/components/Combobox/Combobox.tsx

```diff
@@ -3,16 +3,20 @@ import { useCombobox_unstable } from './useCombobox';
 import { renderCombobox_unstable } from './renderCombobox';
 import { useComboboxStyles_unstable } from './useComboboxStyles';
 import type { ComboboxProps } from './Combobox.types';
+import { useComboboxContextValues } from '../../contexts/useComboboxContext';
+import { useOptionCollection } from '../../utils/useOptionCollection';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
 
 /**
- * Combobox component initial component build
+ * Combobox component: a selection control that allows users to choose from a set of possible options
  */
 export const Combobox: ForwardRefComponent<ComboboxProps> = React.forwardRef((props, ref) => {
-  const state = useCombobox_unstable(props, ref);
+  const optionCollection = useOptionCollection(props.children);
```

Comment:
using this hook directly inside the component breaks our component composition patterns. Our current contract is

```
const MyComponent = (props) => {
  const state = useMyComponentState(props);
  useMyComponentContext(state );
  useMyComponentStyles(state );

  renderMyComponent(state);
}
```

Any hooks that only apply to one component and not others, should not be used here because it's hard to users to know how to recompose the component